hydra-plutus/src/Hydra/Contract/Sha512Example.hs

```diff
@@ -0,0 +1,47 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-# OPTIONS_GHC -fplugin PlutusTx.Plugin #-}
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:defer-errors #-}
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:target-version=1.1.0 #-}
+
+-- | Simple asserting validators that are primarily useful for testing.
+module Hydra.Contract.Sha512Example where
+
+import Hydra.Prelude
+
+import Hydra.Cardano.Api (PlutusScript, pattern PlutusScriptSerialised)
+import PlutusLedgerApi.V3 (ScriptContext (..), ScriptInfo (..), serialiseCompiledCode, unsafeFromBuiltinData)
+import PlutusTx (compile)
+import PlutusTx.Builtins (sha3_512)
+import PlutusTx.Prelude (Eq (..), check)
+
+trivialCheck :: Bool
+trivialCheck =
+  let x = sha3_512 "aaaaaaaa"
+   in x PlutusTx.Prelude.== x
```

Comment:
Are you sure this is not optimized away? A hardcoded, known to be correct hash and pre image would make sure this is indeed run and be slightly better example.

Reply:
There are official test vectors for various SHA algortihms at the bottom of [this page](https://csrc.nist.gov/projects/cryptographic-algorithm-validation-program/secure-hashing).  We use some of these for testing the existing SHA functions in Plutus and I think they include inputs for the functions you're using here as well.

Reply:
No, I assumed as plutus was eager. I agree though.